File: v1/back/java/suggestions/src/test/java/fr/gouv/monprojetsup/suggestions/algos/AffinityEvaluatorTest.java
package fr.gouv.monprojetsup.suggestions.algos;

import fr.gouv.monprojetsup.suggestions.dto.ProfileDTO;
import fr.gouv.monprojetsup.suggestions.data.SuggestionsData;
import fr.gouv.monprojetsup.suggestions.server.SuggestionServer;
import org.junit.BeforeClass;
import org.junit.Test;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.TestInstance;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Lazy;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertTrue;

@TestInstance(TestInstance.Lifecycle.PER_CLASS)
public class AffinityEvaluatorTest {

    @Autowired
    @Lazy
    SuggestionServer server;

    private AffinityEvaluator getTestEvaluator(ProfileDTO profile) {
        Config cfg = new Config();
        return new AffinityEvaluator(profile, cfg);
    }

    @BeforeClass
    @BeforeAll
    public static void prepareData() throws Exception {
        SuggestionsData.load();
    }

    @Test
    public void testDataLoadOk() {
        assertNotNull(SuggestionsData.getDescriptifs());
        /** etc */
        assertFalse(SuggestionsData.getLabels().isEmpty());
    }


    @Test
    public void testGetAffinityEvaluation() {
        //TODO
        assertTrue(true);
    }

    @Test
    public void testGetExplanation() {
        //TODO
        assertTrue(true);
    }

    @Test
    public void testGetCloseTagsSuggestionsOrderedByIncreasingDistance() {
        //TODO
        assertTrue(true);
    }

}

File: app/suggestions/suggestions-server/src/test/fr/gouv/monprojetsup/suggestions/algo/AffinityEvaluatorTest.kt
package fr.gouv.monprojetsup.suggestions.algo

import fr.gouv.monprojetsup.suggestions.dto.ProfileDTO
import org.junit.jupiter.api.Test
import org.springframework.boot.test.context.SpringBootTest

class AffinityEvaluatorTest {

    @Test
    fun `Doit donner un score maximum à une formation proche`() {
        val pf = ProfileDTO(
            "term",
            "générale",
            "court",
            "",
            setOf("44000"),
            setOf(),
            listOf(),
            null,
            listOf(),
            "0"
        )
        /*
        val affinityEvaluator = AffinityEvaluator(pf, Config(), algo)
        val score = affinityEvaluator.getBonusGeographicAffinity(TestData.PARIS_CODE_COMMUNE_INSEE, null)
        assertThat(score).isEqualTo(1.0)*/
    }


}